NeoML/include/NeoML/Dnn/DnnSimpleAPI.h

```diff
@@ -0,0 +1,521 @@
+const int TransformInferenceSame = -2;
+
+NEOML_API CLayerWrapper<CTransformLayer> Transform( int batchLength,
+	int batchWidth, int listSize, int height, int width, int depth, int channel );
+
+NEOML_API CLayerWrapper<CTransposeLayer> Transpose( TBlobDim d1, TBlobDim d2 );
+
+NEOML_API CLayerWrapper<CConcatBatchWidthLayer> ConcatBatchWidth();
+NEOML_API CLayerWrapper<CConcatHeightLayer> ConcatHeight();
+NEOML_API CLayerWrapper<CConcatWidthLayer> ConcatWidth();
+NEOML_API CLayerWrapper<CConcatDepthLayer> ConcatDepth();
+NEOML_API CLayerWrapper<CConcatChannelsLayer> ConcatChannels();
+
+NEOML_API CLayerWrapper<CBatchNormalizationLayer> BatchNormalization(
+	bool isChannelBased, bool isZeroFreeTerm = false, float slowConvergenceRate = 1.0f );
+
+NEOML_API CLayerWrapper<CObjectNormalizationLayer> ObjectNormalization(
+	float epsilon = 1e-5f );
+
+NEOML_API CLayerWrapper<CMultiheadAttentionLayer> MultiheadAttention(
+	int headCount, int hiddenSize, int outputSize, float dropoutRate );
+
+NEOML_API CLayerWrapper<CMatrixMultiplicationLayer> MatrixMultiplication();
+
+NEOML_API CLayerWrapper<CEltwiseSumLayer> Sum();
+NEOML_API CLayerWrapper<CEltwiseMulLayer> Mul();
+NEOML_API CLayerWrapper<CEltwiseNegMulLayer> NegMul();
+NEOML_API CLayerWrapper<CEltwiseMaxLayer> Max();
+
+NEOML_API CLayerWrapper<CGlobalMaxPoolingLayer> GlobalMaxPooling( int maxCount );
+NEOML_API CLayerWrapper<CGlobalMeanPoolingLayer> GlobalMeanPooling();
+
+NEOML_API CLayerWrapper<CLstmLayer> Lstm(
+	int hiddenSize, float dropoutRate, bool isInCompatibilityMode = false );
+
+NEOML_API CLayerWrapper<CGruLayer> Gru( int hiddenSize );
+
+NEOML_API CLayerWrapper<CSubSequenceLayer> SubSequence(
+	int startPos, int length );
+// CSubSequenceLayer with SetReverse()
+NEOML_API CLayerWrapper<CSubSequenceLayer> ReverseSequence();
+
+NEOML_API CLayerWrapper<CArgmaxLayer> Argmax( TBlobDim dim );
+
+// Convolution parameterss along one of the axes.
+struct NEOML_API CConvAxisParams {
+	// FilterSize
+	int Size;
+	// Padding
+	int Padding;
+	// Stride
+	int Stride;
+	// Dilation
+	int Dilation;
+
+	CConvAxisParams() : Size( 1 ), Padding( 0 ), Stride( 1 ), Dilation( 1 ) {}
+	explicit CConvAxisParams( int size, int padding = 0, int stride = 1, int dilation = 1 ) :
+		Size( size ),
+		Padding( padding ),
+		Stride( stride ),
+		Dilation( dilation )
+	{
+	}
+};
+
+NEOML_API CLayerWrapper<CConvLayer> Conv( int filterCount,
+	const CConvAxisParams& heightParams, const CConvAxisParams& widthParams,
+	bool isZeroFreeTerm = false );
+
+// Convolution along width dimension, i.e:
+//	filter height = 1
+//	stride height = 1
+//	dilation height = 1
+//	padding height = 0
+inline CLayerWrapper<CConvLayer> ConvByWidth( int filterCount,
+	int size, int padding = 0, int stride = 1, int dilation = 1,
+	bool isZeroFreeTerm = false )
+{
+	return Conv( filterCount, CConvAxisParams(),
+		CConvAxisParams( size, padding, stride, dilation ), isZeroFreeTerm );
+}
+
+// Convolution along height dimension, i.e:

```

Comment:
Is it really helpful?

Reply:
Yes, at least for ConvByWidth. It is convenient to use this func in network for text processing, like BERT